ntbs-service/DataMigration/ImportValidator.cs

```diff
@@ -89,48 +128,54 @@ public ImportValidator(IImportLogger logger, IReferenceDataRepository referenceD
             {
                 notification.TestData.HasTestCarriedOut = false;
             }
+        }
 
-            if (ValidateObject(notification.ContactTracing).Any())
+        private async Task CleanContactTracingData(Notification notification, PerformContext context, int runId)
+        {
+            if (!ValidateObject(notification.ContactTracing).Any())
             {
-                const string message = "invalid contact tracing figures. " +
-                                       "The notification will be imported without contact tracing data.";
-                await _logger.LogNotificationWarning(context, runId, notification.LegacyId, message);
-                notification.ContactTracing = new ContactTracing();
+                // If there are no issues in the contact tracing
+                return;
             }
-        }
 
-        private static void TrimAndCleanStringProperties(object entity)
-        {
-            // Clean strings properties of this object
-            entity?.GetType()
-                .GetProperties()
-                .Where(prop => prop.PropertyType == typeof(string)
-                               && prop.CanWrite
-                               && (Attribute.IsDefined(prop, typeof(RegularExpressionAttribute))
-                                   || Attribute.IsDefined(prop, typeof(MaxLengthAttribute))))
-                .ForEach(prop => prop.SetValue(entity, TrimAndCleanString((string)prop.GetValue(entity))));
+            var messages = new List<string>();
+            var lostData = new List<string>();
 
-            // Clean strings properties of this object's entity children
-            entity?
-                .GetType()
-                .GetProperties()
-                .Where(prop => Attribute.IsDefined(prop, typeof(ValidationChildAttribute)))
-                .ForEach(prop => TrimAndCleanStringProperties(prop.GetValue(entity)));
+            // The order we clean properties is important, because if we overwrite one value (eg. AdultsScreened) with
+            // null then we also need to overwrite the values that depend on it (eg. AdultsActiveTB)
```

Comment:
Maybe "that are necessarily lower than it" is slightly clearer than "that depend on it"?

Reply:
Yep!